Duplicati/Library/Backend/Storj/StorjBackend.cs

```diff
@@ -0,0 +1,370 @@
+﻿using Duplicati.Library.Interface;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using uplink.NET.Interfaces;
+using uplink.NET.Models;
+using uplink.NET.Services;
+
+namespace Duplicati.Library.Backend.Storj
+{
+    public class Storj : IStreamingBackend
+    {
+        private const string STORJ_AUTH_METHOD = "storj-auth-method";
+        private const string STORJ_SATELLITE = "storj-satellite";
+        private const string STORJ_API_KEY = "storj-api-key";
+        private const string STORJ_SECRET = "storj-secret";
+        private const string STORJ_SECRET_VERIFY = "storj-secret-verify";
```

Comment:
How is this (as well as `TARDIGRADE_SECRET_VERIFY`) used in the C# code?  Is it necessary?

Reply:
It is not used in code, it is used in the UI. But I got a warning about that parameter not being allowed on that backend so I added it.
I think I have to remove it somewhere so that it is not passed to the backend. But I'm not quite sure if it will be available in the ui-verificator then. Can you guide me a bit here?
It is used in EditUriBuiltins.js for the parser, the builders and the validators. Technically it is only necessary in the validator. Could I simply remove it in the other two occurencies? The parameter should then not be transfered to the backend and I could remove it there then.

Reply:
It looks like the password verification is a new feature.  Was there a reason why it was added?  I don't think any of the other backends have a verification field, and the "Test Connection" button should give sufficient feedback about the credentials.

Reply:
The password verification here is new, yes, and it is really necessary. The password here is not used to access Storj but to encrypt the files before uploading. If you were to make a typo here, it would still work - but you would not be able to successfully decrypt your data afterwards. So I've added a check-field to verify that the password is exactly what the user wanted.
I just don't really need that secret-verify-field in the backend, as the check is already done in the frontend.

Reply:
Thanks for the explanation, I understand the purpose now.  I ran some tests by removing the 4 lines in the C# code, and I encountered the warnings you described.  If I then remove the following lines (which are involved in building the destination URI), I no longer get the warnings.  Can you try this and confirm if you see the same?

https://github.com/duplicati/duplicati/blob/a2a7e4b7ba1963192d636c53dc01cf6d969db416/Duplicati/Server/webroot/ngax/scripts/services/EditUriBuiltins.js#L804

https://github.com/duplicati/duplicati/blob/a2a7e4b7ba1963192d636c53dc01cf6d969db416/Duplicati/Server/webroot/ngax/scripts/services/EditUriBuiltins.js#L826

Reply:
Yes, removing these lines looks good - the check is still working in the frontend but we don't get the value in the backend. So I could remove the secret-verify-parameter in the backend completely.